### PR TITLE
Update README to address typos and minor content changes

### PR DESCRIPTION
**Pull request by sp717**
*Issue #, if available:*

*Description of changes:*
- Fix multiple misspelled words and typos. 
- Add OAEPWithSHA1AndMGF1Padding to list of Ciphers.
- Add Go to the manual installation requirements. 
- Update description of SecureRandom.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by geedo0 — APPROVED** [review]
Good eye, thanks!

File: README.md [hunk 1]
```diff
@@ -107,7 +108,7 @@ Notable differences between ACCP and ACCP-FIPS:
 
 ACCP-FIPS is only supported on the following platforms:
 
-| Platfrom | FIPS support since version |
+| Platform | FIPS support since version |
```
Comment (WillChilds-Klein) [inline README.md:111]: while we're in here, let's update our description of SecureRandom on L85 as AWS-LC [no longer defaults to jitter-based entropy](https://github.com/aws/aws-lc/blob/b2bbd9c287d3fff4e209892d342ea3b0d562477e/CMakeLists.txt#L25). I'd suggest something more generic, deferring documentation to AWS-LC:

```
* ACCP's SecureRandom uses [AWS-LC's DRBG implementation](https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/rand/rand.c).
```

File: README.md [hunk 2]
```diff
@@ -222,6 +223,9 @@ Building this provider requires a 64 bit Linux or MacOS build system with the fo
 * C++ build chain
 * [lcov](http://ltp.sourceforge.net/coverage/lcov.php) for coverage metrics
 * [gcovr](https://gcovr.com/en/stable/) for reporting coverage metrics in CodeBuild
+* [Go](https://golang.org/dl/) 1.18 or later is required. If not found by
```
Comment (alexw91) [inline README.md:226]: This isn't currently documented in AWS-LC's documentation, but Go v1.20 or later should be used when possible. 
ninja-build: fine with Go 1.18
ninja-build run_test: sorry, need Go 1.20

I'd change this line to `1.18 or later is the minimum required version to build AWS-LC, 1.20 or later is needed in order to run AWS-LC's test suite.`

File: README.md (hunk 1 above)
Comment (sp717) [reply README.md:111]: Sounds good, just changed in latest commit.

File: README.md (hunk 2 above)
Comment (sp717) [reply README.md]: Updated!

**Review by alexw91 — APPROVED** [review]

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**